### PR TITLE
Add authn support

### PR DESCRIPTION
**Pull request by diclophis**
Add rudimentary support for `/authn` endpoint

### CODE REVIEW

**Comment by diclophis** [comment]
@wvanbergen / @andremedeiros I could use some help on this PR in terms of, writing a spec for my new API endpoint, as well as any direction / advice / considerations for adding `authn` support to this lib... Thanks!

**Comment by wvanbergen** [comment]
Thanks for contributing! I am not too aware of the authn endpoint myself because we don't use it, but I am happy to merge a PR that adds support for it.

File: lib/oktakit/client/authentication.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+module Oktakit
+  class Client
+    module Authentication
+      # Authenticate User
+      #
+      # @param username [String]. Username for authn request
+      # @param password [String]. Password for authn request
+      # @param mfa [String]. Multi-Factor Token for authn request (only first found "token:software:totp" supported)
+      # @example
+      #   Oktakit.authn('user', 'pass', 123456)
```
Comment (wvanbergen) [inline lib/oktakit/client/authentication.rb]: Is MFA enabled for everyone? Or should the argument be optional? If it's optional, I rather use a keyword argument.

**Comment by diclophis** [comment]
closing this as our use case is no longer required

**Pull request closed without merging**